core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -268,6 +315,18 @@ func (ec *ethConfirmer) findEthTxAttemptsRequiringReceiptFetch() (attempts []mod
 	return
 }
 
+func (ec *ethConfirmer) getNonceForLatestBlock(ctx context.Context, from gethCommon.Address) (nonce uint64, err error) {
+	ctx, cancel := eth.DefaultQueryCtx(ctx)
+	defer cancel()
+	at, err := ec.ethClient.NonceAt(ctx, from, nil)
+
+	if ctx.Err() != nil {
```

Comment:
This check feels in the wrong place

Reply:
hm. it's similar to others, e.g.
```
if ctx.Err() != nil {
		return nil
	} else if err != nil {
		return err
	}
```
https://github.com/smartcontractkit/chainlink/blob/develop/core/services/head_tracker.go#L223

Reply:
seems similar to others like:
https://github.com/smartcontractkit/chainlink/blob/develop/core/services/head_tracker.go#L223

Though i can see others usually break the loop or return nil.. i'll try a similar thing